tests/test_transaction_db.py

```diff
@@ -729,21 +734,27 @@ def test_user_preferences(self):
         self.assertEqual(user_id_2, prefs['user_id'])
         self.assertEqual('de', prefs['report_language'])
 
-    @unittest.expectedFailure
+    # TODO
+    @unittest.skip("Key validation does not work for whatever reason.")
     def test_user_preferences_invalid_keys(self):
-        """test that upon providing an invalid preference key for the
-        preferences system, a TransactionDBException is thrown"""
+        """Test that invalid preference key raises `TransactionDBException`."""
         engine = temp_db.get_temp_db()
         t_db = TransactionDB(engine)
         user_id_1 = t_db.add_user('Pere1', 'pwd')
 
-        t_db.set_user_preferences(user_id_1, {'foo': 'bar'})
+        with self.assertRaises(TransactionDBException):
+            t_db.set_user_preferences(user_id_1, {'foo': 'bar'})
 
+    # TODO
+    @unittest.skip("Key validation does not work for whatever reason.")
```

Comment:
I don't understand why this test is failing. It was even before I started working on the project and chaging things. It should get rejected by SQLAlchemy by default, right?